relay-general/src/protocol/span.rs

```diff
@@ -35,6 +35,12 @@ pub struct Span {
     /// The status of a span
     pub status: Annotated<SpanStatus>,
 
+    /// Arbitrary tags on a span, like on the top-level event. Will maybe be indexed in the future.
```

Comment:
```suggestion
    /// Arbitrary tags on a span, like on the top-level event.
```

I'd not talk about indexing here. This is not Relay's concern, and we won't come back update such kind of comments when we do indexing.

Reply:
I don't want to leave this without some kind of guideline as to when to use data vs tags.

Reply:
I think the difference is semantic (and data type). We don't go document for each field whether it gets indexed or not by a completely different part of the system.

Tags: key-value pairs, all strings, typically short in length, used to distinguish/group spans. E.g.: `"color": "blue"`
Data: string-to-anything pairs, used to store arbitrary data that goes along with a span. E.g.: `"context": {"request_id": "deadbeef", "metered": true}`

Reply:
I agree with Rodolfo here. Since this is intended to eventually be consumed by customers, you could talk about whether something is available in search. However, I would not make statements about this just yet, since there is an open conversion about this in a different place right now.

Reply:
I don't agree that there is any semantic difference left between span.data and span.tags if you choose to leave out how the data would potentially be indexed in the future. What sense does it make to introduce two namespaces depending on whether the strings are short or of lower cardinality? That's not useful advice, given that tags in events are not necessarily low cardinality at all, and at best it's put out of context given that the length restriction on tags is directly resulting from technical restrictions in indexing.

>We don't go document for each field whether it gets indexed or not by a completely different part of the system.

Whenever we don't that's how we end up with confusing duplicate concepts like extra vs context. In the case of event tags we do say that it's their only purpose whenever asked because there is no reason for them to exist when extra is there as well.

But most of the time we actually do because we document what kind of content you can search for in Discover: https://docs.sentry.io/product/discover-queries/query-builder/#filter-by-search-conditions

>I would not make statements about this just yet

*That* I can agree with though I feel like a "maybe" already makes very little promises.

Reply:
Just fyi, span.data and span.tags are presented differently in the UI.